<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qlfoframe</class>
 <widget class="QFrame" name="qlfoframe">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>663</width>
    <height>360</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>LFO</string>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>601</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QLabel" name="m_lfo_main_label">
      <property name="font">
       <font>
        <pointsize>11</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>LFO Data Editor</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="m_pattern_label">
      <property name="font">
       <font>
        <italic>true</italic>
       </font>
      </property>
      <property name="text">
       <string>Pattern #xx</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="m_measures_check_box">
      <property name="toolTip">
       <string>Use the measure length, rather than pattern length,
as the length of the LFO period.</string>
      </property>
      <property name="text">
       <string>Use Measure (vs Length)</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QCheckBox" name="m_multiply_check_box">
      <property name="toolTip">
       <string>If checked, multiply existing event values by the
waveform, rather than forcing the events to
match the waveform.</string>
      </property>
      <property name="text">
       <string>Multiply</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="m_wave_type_group">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>40</y>
     <width>291</width>
     <height>261</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>Wave Type</string>
   </property>
   <widget class="QWidget" name="radioButtonLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>251</width>
      <height>221</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="m_wave_layout">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>12</number>
     </property>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_none">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>&amp;None</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_sine">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Sine</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_saw">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Sawtooth</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_revsaw">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Re&amp;verse Sawtooth</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_triangle">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Triangle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_exp">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>E&amp;xponential Rise</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_revexp">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Exponential Fall</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="m_radio_wave_dc">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>DC Only</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QWidget" name="buttonLayoutWidget">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>304</y>
     <width>296</width>
     <height>34</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <property name="spacing">
     <number>3</number>
    </property>
    <item>
     <widget class="QPushButton" name="m_button_reset">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>28</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Resets the pattern events to the original set
or to the last &quot;locked&quot; set of events.</string>
      </property>
      <property name="text">
       <string>&amp;Reset</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="m_button_lock">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>28</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Makes the current, modified events official and
non-resettable. Further modifications can be
applied.</string>
      </property>
      <property name="text">
       <string>&amp;Lock</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="m_button_close">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>28</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Close the LFO frame. If the pattern is modified, raise
the modify flag for the MIDI tune.</string>
      </property>
      <property name="text">
       <string>&amp;Close</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="sliderLayoutWidget">
   <property name="geometry">
    <rect>
     <x>18</x>
     <y>53</y>
     <width>323</width>
     <height>281</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="v_value_layout" stretch="0,0,0">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QLabel" name="m_value_label">
        <property name="text">
         <string>DC Offset</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="m_value_slider">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>DC offset to add to modulating waveform.</string>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::NoTicks</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="m_value_text">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>100.0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="v_range_layout">
      <item>
       <widget class="QLabel" name="m_range_label">
        <property name="text">
         <string>Depth</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="m_range_slider">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Depth of modulation.</string>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::NoTicks</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="m_range_text">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>100.0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="v_speed_layout">
      <item>
       <widget class="QLabel" name="m_speed_label">
        <property name="text">
         <string>Periods</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="m_speed_slider">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Number of periods per pattern. Subject to anti-aliasing.</string>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::NoTicks</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="m_speed_text">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>100.0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="v_phase_layout">
      <item>
       <widget class="QLabel" name="m_phase_label">
        <property name="text">
         <string>Phase</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="m_phase_slider">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Phase of the LFO, 0 to 1 is 0 to 360 degrees.</string>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="invertedControls">
         <bool>false</bool>
        </property>
        <property name="tickPosition">
         <enum>QSlider::NoTicks</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="m_phase_text">
        <property name="minimumSize">
         <size>
          <width>72</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>360.0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>m_measures_check_box</tabstop>
  <tabstop>m_range_slider</tabstop>
  <tabstop>m_range_text</tabstop>
  <tabstop>m_speed_slider</tabstop>
  <tabstop>m_speed_text</tabstop>
  <tabstop>m_phase_slider</tabstop>
  <tabstop>m_phase_text</tabstop>
  <tabstop>m_radio_wave_none</tabstop>
  <tabstop>m_radio_wave_sine</tabstop>
  <tabstop>m_radio_wave_saw</tabstop>
  <tabstop>m_radio_wave_revsaw</tabstop>
  <tabstop>m_radio_wave_triangle</tabstop>
  <tabstop>m_radio_wave_exp</tabstop>
  <tabstop>m_radio_wave_revexp</tabstop>
  <tabstop>m_button_reset</tabstop>
  <tabstop>m_button_close</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
